app/src/main/jni/moonlight-core/callbacks.c

```diff
@@ -80,7 +80,7 @@ Java_com_limelight_nvstream_jni_MoonBridge_init(JNIEnv *env, jclass clazz) {
     BridgeDrStartMethod = (*env)->GetStaticMethodID(env, clazz, "bridgeDrStart", "()V");
     BridgeDrStopMethod = (*env)->GetStaticMethodID(env, clazz, "bridgeDrStop", "()V");
     BridgeDrCleanupMethod = (*env)->GetStaticMethodID(env, clazz, "bridgeDrCleanup", "()V");
-    BridgeDrSubmitDecodeUnitMethod = (*env)->GetStaticMethodID(env, clazz, "bridgeDrSubmitDecodeUnit", "([BIIIIJJ)I");
+    BridgeDrSubmitDecodeUnitMethod = (*env)->GetStaticMethodID(env, clazz, "bridgeDrSubmitDecodeUnit", "([BIIIISJJ)I");
```

Comment:
The latency is *unsigned* 16bit integer, can't use signed 16bit type here. You can either use 16bit unsigned integer ([jni types](https://docs.oracle.com/javase/8/docs/technotes/guides/jni/spec/types.html)), or expand and pass it as 32bit signed integer. Same thing in java code.

Reply:
Thanks for catching that, I made the update to the unsigned char/jchar everywhere.